pkg/virt-operator/install-strategy/strategy.go

```diff
@@ -348,6 +415,261 @@ func addOperatorLabel(objectMeta *metav1.ObjectMeta) {
+				err := clientset.RbacV1().ClusterRoles().Delete(cr.Name, deleteOptions)
+				if err != nil {
+					expectations.ClusterRole.DeletionObserved(kvkey, key)
+					log.Log.Errorf("Failed to delete cr %+v: %v", cr, err)
+					return err
+				}
+			}
+		} else if !ok {
+			log.Log.Errorf("Cast failed! obj: %+v", obj)
+			return nil
+		}
+	}
+
+	objects = stores.RoleBindingCache.List()
+	for _, obj := range objects {
+		if rb, ok := obj.(*rbacv1.RoleBinding); ok && rb.DeletionTimestamp == nil {
+			if key, err := controller.KeyFunc(rb); err == nil {
+				expectations.RoleBinding.AddExpectedDeletion(kvkey, key)
+				err := clientset.RbacV1().RoleBindings(kv.Namespace).Delete(rb.Name, deleteOptions)
+				if err != nil {
+					expectations.RoleBinding.DeletionObserved(kvkey, key)
+					log.Log.Errorf("Failed to delete rb %+v: %v", rb, err)
+					return err
+				}
+			}
+		} else if !ok {
+			log.Log.Errorf("Cast failed! obj: %+v", obj)
+			return nil
+		}
+	}
+
+	objects = stores.RoleCache.List()
+	for _, obj := range objects {
+		if role, ok := obj.(*rbacv1.Role); ok && role.DeletionTimestamp == nil {
+			if key, err := controller.KeyFunc(role); err == nil {
+				expectations.Role.AddExpectedDeletion(kvkey, key)
+				err := clientset.RbacV1().Roles(kv.Namespace).Delete(role.Name, deleteOptions)
+				if err != nil {
+					expectations.Role.DeletionObserved(kvkey, key)
+					log.Log.Errorf("Failed to delete role %+v: %v", role, err)
+					return err
+				}
+			}
+		} else if !ok {
+			log.Log.Errorf("Cast failed! obj: %+v", obj)
+			return nil
+		}
+	}
+
+	objects = stores.ServiceAccountCache.List()
+	for _, obj := range objects {
+		if sa, ok := obj.(*corev1.ServiceAccount); ok && sa.DeletionTimestamp == nil {
+			if key, err := controller.KeyFunc(sa); err == nil {
+				expectations.ServiceAccount.AddExpectedDeletion(kvkey, key)
+				err := clientset.CoreV1().ServiceAccounts(kv.Namespace).Delete(sa.Name, deleteOptions)
+				if err != nil {
+					expectations.ServiceAccount.DeletionObserved(kvkey, key)
+					log.Log.Errorf("Failed to delete serviceaccount %+v: %v", sa, err)
+					return err
+				}
+			}
+		} else if !ok {
+			log.Log.Errorf("Cast failed! obj: %+v", obj)
+			return nil
+		}
+	}
+
+	scc := clientset.SecClient()
+	for _, sccPriv := range strategy.customSccPrivileges {
+		privSccObj, exists, err := stores.SCCCache.GetByKey(sccPriv.TargetScc)
+		if !exists {
+			return nil
+		} else if err != nil {
+			return err
+		}
+
+		privScc, ok := privSccObj.(*secv1.SecurityContextConstraints)
+		if !ok {
+			return fmt.Errorf("couldn't cast object to SecurityContextConstraints: %+v", privSccObj)
+		}
+		privSccCopy := privScc.DeepCopy()
+
+		modified := false
+		users := privSccCopy.Users
+		for _, acc := range sccPriv.ServiceAccounts {
+			removed := false
+			users, removed = remove(users, acc)
+			modified = modified || removed
+		}
+
+		if modified {
+			privSccCopy.Users = users
+			_, err = scc.SecurityContextConstraints().Update(privSccCopy)
```

Comment:
I know the Update was already there before but it concerns me a little bit. Since we don't create the object, we might erase non-default fields on newer openshift versions. Can we use patch here?

Reply:
If the object changed in the meantime yes. But we are updating an object which can contain fields which our struct does not even know about. It would remove all unknown fields. That is not an issue if we create and maintain an object, because we would only add fields we know about in the first place.

Reply:
yup, fixed